packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -107,13 +114,38 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
           this.setState({ isOpen: false });
         }
 
-        this.props.onDidDismiss && this.props.onDidDismiss(evt);
+        const didComponentUnmount = this.ref.current === null;
+
+        if (didComponentUnmount === false) {
+          /**
+           * If the component is unmounted, we do not
+           * want to call the onDidDismiss handler to
+           * prevent a React state update on an unmounted component.
+           */
+          this.props.onDidDismiss && this.props.onDidDismiss(evt);
+        }
       });
     }
 
     componentDidUpdate(prevProps: IonicReactInternalProps<PropType>) {
       const node = this.ref.current! as HTMLElement;
-      attachProps(node, this.props, prevProps);
+      const propsToPass = Object.keys(this.props).reduce((acc, name) => {
+        if (!excludedOverlayEvents.includes(name)) {
+          /**
+           * Maintains all props except for the event handlers
+           * that are manually handled in the componentDidMount.
+           */
+          (acc as any)[name] = (this.props as any)[name];
+        }
+        return acc;
+      }, {});
```

Comment:
Why is this needed?

Reply:
This change prevents `onDidDismiss` being invoked after the component is unmounted, resulting in the React error:
> Can't perform a React state update on an unmounted component.

There are multiple conditions that cause this, from either us trying to handle the method in the event listener after the component has unmounted (`didComponentUnmount` prevents this) or from the `attachProps` util (which also adds an event listener and can execute after the component has unmounted).